larq_zoo/sota/quicknet_xl.py

```diff
@@ -0,0 +1,179 @@
+from typing import Optional, Sequence, Tuple
+
+import larq as lq
+import tensorflow as tf
+from zookeeper import Field, factory
+
+from larq_zoo.core import utils
+from larq_zoo.core.model_factory import ModelFactory
+from larq_zoo.sota.quicknet import LCEFirstLayer
+
+
+def squeeze_and_excite(inp: tf.Tensor, strides: int = 1, r: int = 16):
+    """Squeeze and Excite as per [Squeeze-and-Excitation Networks](https://arxiv.org/abs/1709.01507)"""
+    C = inp.get_shape().as_list()[-1]
+
+    out = utils.global_pool(inp)
+    out = tf.keras.layers.Dense(
+        C // r,
+        activation="relu",
+        kernel_initializer="he_normal",
+        use_bias=False,
+        kernel_regularizer=tf.keras.regularizers.l2(1e-5),
+    )(out)
+    outmult = 2 if strides == 2 else 1
+    out = tf.keras.layers.Dense(
+        C * outmult,
+        activation="sigmoid",
+        kernel_initializer="he_normal",
+        use_bias=False,
+        kernel_regularizer=tf.keras.regularizers.l2(1e-5),
+    )(out)
+    return tf.reshape(out, [-1, 1, 1, C * outmult])
```

Comment:
Can we reuse the [implementation from `QuickNetLarge`](https://github.com/larq/zoo/blob/master/larq_zoo/sota/quicknet_large.py#L12-L32) here instead?